qemu/tests/usb_storage.py

```diff
@@ -193,56 +195,83 @@ def _check_io_size_option(min_io_size="512", opt_io_size="0"):
-
-    if params.get("check_io_size_option") == "yes":
-        error_context.context("Check usb min/opt io_size option", logging.info)
-        _check_io_size_option("0", "0")
-        # Guest can't recognize correct value which we set now,
-        # So comment these test temporary.
-        # _check_io_size_option("1024", "1024")
-        # _check_io_size_option("4096", "4096")
+    hotplug_unplug = (params["with_hotplug_unplug"] == "yes")
+    repeat_times = int(params.get("usb_repeat_times", "1"))
+    for rt in range(1, repeat_times+1):
+        disk_hotplugged = []
+        if hotplug_unplug:
+            error_context.context("Hotplug the %s times." % rt, logging.info)
+            image_name = params.objects("images")[-1]
+            image_params = params.object_params(image_name)
+            devices = vm.devices.images_define_by_params(image_name,
+                                                         image_params,
+                                                         'disk', None,
+                                                         False, None)
+            for dev in devices:
+                ret = vm.devices.simple_hotplug(dev, vm.monitor)
+                if ret[1] is False:
+                    test.fail("Failed to hotplug device '%s'. Output:\n%s"
+                              % (dev, ret[0]))
+            disk_hotplugged.append(devices[-1])
+
+        error_context.context("Check usb device information in monitor",
+                              logging.info)
+        output = str(vm.monitor.info("usb"))
+        if "Product QEMU USB MSD" not in output:
+            logging.debug(output)
+            test.fail("Could not find mass storage device")
+
+        error_context.context("Check usb device information in guest",
+                              logging.info)
+        session = _login()
+        output = session.cmd(params["chk_usb_info_cmd"])
+        # No bus specified, default using "usb.0" for "usb-storage"
+        for i in params["chk_usb_info_keyword"].split(","):
+            _verify_string(i, output, [i])
+        session.close()
+        _do_io_test_guest()
+
+        # this part is linux only
+        if params.get("check_serial_option") == "yes":
+            error_context.context("Check usb serial option", logging.info)
+            serial = str(uuid.uuid4())
+            regex_str = r'usb-storage.*?serial = "(.*?)"\s'
+            _check_serial_option(serial, regex_str, serial)
+
+            logging.info("Check this option with some illegal string")
+            logging.info("Set usb serial to a empty string")
+            # An empty string, ""
+            serial = "EMPTY_STRING"
+            regex_str = r'usb-storage.*?serial = (.*?)\s'
+            _check_serial_option(serial, regex_str, '""')
+
+            logging.info("Leave usb serial option blank")
+            serial = "NO_EQUAL_STRING"
+            regex_str = r'usb-storage.*?serial = (.*?)\s'
```

Comment:
is this `regex_str` same as LINE 245?

Reply:
Have deleted the duplicate one, Line250. Thanks!